src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -525,34 +526,29 @@ private void WriteBits(byte value, int count)
             // and 230ns on 5V. (PWeh on page 49/52 and Figure 25 on page 58)
 
             _controller.Write(_enablePin, PinValue.High);
-            DelayMicroseconds(1);
+            DelayMicroseconds(1, allowThreadYield: false);
             _controller.Write(_enablePin, PinValue.Low);
         }
 
-        private void DelayMicroseconds(int microseconds, bool checkBusy = true)
+        private void DelayMicroseconds(int microseconds, bool allowThreadYield = true,  bool checkBusy = true)
         {
-            _stopwatch.Restart();
+            DelayHelper.DelayMicroseconds(microseconds, allowThreadYield);
 
-            // While we could check for the busy state it isn't currently practical.
-            // Most commands need a maximum of 37μs to complete. Reading the busy flag
-            // alone takes ~200μs (on the Pi). Prepping the pins and reading once can take
-            // nearly a millisecond, which clearly is not going to be performant.
-            // 
-            // Leaving the flag here to make sure calling code is describing when it
-            // cannot actually check the busy state should we be able to get performance
+            // While we could check for the busy state it isn't currently practical. Most
+            // commands need a maximum of 37μs to complete. Reading the busy flag alone takes
+            // ~200μs (on the Pi) if going through the software driver. Prepping the pins and
+            // reading once can take nearly a millisecond, which clearly is not going to be
+            // performant.
+
+            // Leaving the flag here to make sure calling code is describing when it cannot
+            // actually check the busy state should we be able to get performance
             // to the point where checking would be a net benefit.
 
             // Note that on a Raspberry Pi 3B+ we average about 1.5μs when delaying for one μs.
             // 
             // SpinWait currently spins to approximately 1μs before it will yield the thread.
             // Thread.Yield() takes around 1.2μs. This gives us a fidelity of microseconds to
             // microseconds + 1.4.
```

Comment:
Should this part of the comment about `SpinWait`, and `we average about 1.5μs when delaying for one μs`, be removed or modified?

Reply:
Yeah, I'll remove it.